features/product/viewing_products/viewing_product_price_on_products_list.feature

```diff
@@ -33,3 +33,13 @@ Feature: Viewing a product price on products list
         When I browse products from taxon "T-Shirts"
         Then I should see the product "T-Shirt watermelon" with price "$19.00"
         And I should see "T-Shirt watermelon" product not discounted on the list
+
+    @ui @api
+    Scenario: Viewing a product with a positioned default variant
+        Given the store has a product "T-Shirt watermelon" priced at "$19.00"
+        And the product "T-Shirt watermelon" has also an "Extra Large" variant at position -1
+        And this product's price is "$10.00"
```

Comment:
It will need a new way of setting the price, because here it ask the to the DefaultProductVariantResolver what is the available variant and set the price...

Reply:
Yes, that's what I noted in my comment below.

Reply:
Check the methods that use `createProductVariant` in `ProductContext`.
There are methods like `theProductHasVariants`, `theProductHasVariantWithCode`, `theProductHasVariantPricedAt`, `theProductHasVariantAtPosition`.

Reply:
A combination of `theProductHasVariantAtPosition` and `thisVariantIsAlsoPricedAtInChannel` should do the job.

Reply:
Ah yes you are right, I did not pay attention to this method `thisVariantIsAlsoPricedAtInChannel` ! Thx 😉

Reply:
@vvasiloi Not possible, because it adds a new channel pricing instead of using the existing one : 

```
An exception occurred while executing a query: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '24-WEB-US' for key 'sylius_channel_pricing.product_variant_channel_idx'
```

Reply:
Then maybe modify the method to check if it already has channel pricing before adding a new one?

Reply:
That's what I expected it to do tbh...